include/tsl/ordered_hash.h

```diff
@@ -1098,6 +1098,44 @@ class ordered_hash : private Hash, private KeyEqual {
     return 1;
   }
 
+  /**
+   * Clear a bucket without touching the container holding the values.
+   */
+  void clear_bucket(typename buckets_container_type::iterator it) {
+    tsl_oh_assert(it != m_buckets_data.end());
+    it->clear();
+    backward_shift(std::size_t(std::distance(m_buckets_data.begin(), it)));
+  }
+
+  /**
+   * Remove all entries for which the given predicate matches.
+   */
+  template <class Predicate>
+  size_type erase_if(Predicate &pred) {
+    using const_ref = typename values_container_type::const_reference;
```

Comment:
Any reason why needing const reference?

From https://en.cppreference.com/w/cpp/container/vector/erase2 

> pred 	- 	unary predicate which returns ​true if the element should be erased.
The expression pred(v) must be convertible to bool for every argument v of type (possibly const) T, regardless of [value category](https://en.cppreference.com/w/cpp/language/value_category), and must not modify v. Thus, a parameter type of T&is not allowed, nor is T unless for T a move is equivalent to a copy (since C++11). ​

A `pred` that takes a mutable reference should still work.

Reply:
This would allow the user to change the key, right? STL maps use `std::pair<const K, V>` as value type but with the ordered map use `std::pair<K, V>`. My goal was to prevent this.

Reply:
I guess one can also leave it in the user's responsibility not to change the key. I don't see any drawbacks of the current implementation, though. But then this is C++ and there might be issues...

Reply:
Effectively, probably better to keep it like that.